setup.py

```diff
@@ -1,11 +1,18 @@
 #!/usr/bin/env python
-from distutils.core import setup
-
-# hack for utf8 long_description support
+from setuptools import setup
 import sys
-reload(sys).setdefaultencoding("UTF-8")
+import setuptools
+
+version='0.8'
+EXTRAS_REQUIRE = {}
+INSTALL_REQUIRES = []
+
 
-version='0.7'
+if int(setuptools.__version__.split(".", 1)[0]) < 18:
+    if sys.version_info[0:2] < (3, 0):
+        INSTALL_REQUIRES.append("future")
```

Comment:
I'm not a fan of `future` module because it can break other packages just because it is installed/imported, see e.g. https://github.com/rkern/line_profiler/issues/12. How hard would it be to replace it with something else? Is it even used?